common/doordeck.js

```diff
@@ -1,51 +1,61 @@
-import { AUTH_TOKEN } from './constants'
+import {AUTH_TOKEN, BASE_URL, DEFAULT_BASE_URL} from './constants'
 import ephemaralKeyGenerator from './services/ephemeralKeyGenerator'
 import certificate from './services/certificate'
 import libSodium from 'libsodium-wrappers'
 import device from "./services/deviceOperation";
 
 const isLoaded = async function (authToken) {
-  if (authToken === getStoredAuthToken()) {
-    if (ephemaralKeyGenerator.retrieveSavedKeys() !== null) {
-      return (await certificate.retrieveSavedCert()) !== null;
+    if (authToken === getStoredAuthToken()) {
+        if (ephemaralKeyGenerator.retrieveSavedKeys() !== null) {
+            return (await certificate.retrieveSavedCert()) !== null;
+        } else return false
     } else return false
-  } else return false
-}
-
-const doordeckInit = function (authToken) {
-  return new Promise (function(resolve, reject) {
-    libSodium.ready.then(async function () {
-      if ((await isLoaded())) resolve({state: 'success', message: 'Doordeck is already initialised.'})
-      if (authToken !== null && authToken !== undefined) {
-        storeAuthToken(authToken)
-        ephemaralKeyGenerator.generateKeys().then(keys => {
-          certificate.getCertificate(keys).then(response => {
-            resolve(response)
-          }, fail => {
-            reject(fail)
-          })
+}
+
+const doordeckInit = function (authToken, baseUrl) {
```

Comment:
Do we need to default it here for backwards compatibility?

Reply:
@gunhansancar was talking about increasing a major version to allow the breaking changes.
This is also an option @gunhansancar, what do you think?
I think it's a good idea

Reply:
If we remove `baseUrl` from other functions that ends up with a signature change which in [semantic versioning](https://semver.org/) rules requires major version change but of course it is up to us to follow that.

Reply:
I've recently added it, but it's added in all except one endpoint, I think.
I believe we can update the major anyway and have this work done properly